schemas/string_interpolation/transforms/generate_date_range.json

```diff
@@ -0,0 +1,74 @@
+{
+  "$id": "https://eq.ons.gov.uk/string_interpolation/transforms/generate_date_range.json",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "generate_date_range": {
+    "description": "A transform for generating the start and end date of a date range from a reference date, offset number of weeks, number of days in range and an optional first day of the week.",
+    "type": "object",
+    "properties": {
+      "transform": {
+        "type": "string",
+        "enum": ["generate_date_range", "format_date_range_pair"]
+      },
+      "arguments": {
+        "type": "object",
+        "properties": {
+          "reference_date": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
+              {
+                "$ref": "https://eq.ons.gov.uk/common_definitions.json#/date_value"
+              }
+            ]
+          },
+          "offset_full_weeks": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
+              {
+                "type": "integer"
+              }
+            ]
+          },
+          "days_in_range": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
+              {
+                "type": "integer",
+                "minimum": 1
+              }
+            ]
+          },
+          "first_day_of_week": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
+              {
+                "type": "string",
+                "default": "monday",
+                "enum": [
+                  "monday",
+                  "tuesday",
+                  "wednesday",
+                  "thursday",
+                  "friday",
+                  "saturday",
+                  "sunday"
+                ]
```

Comment:
In v2, we used abbreviation for days i.e `MO`, `TU`, in the proposal I did for v3 dynamic option, I used abbrev like `MON`, `TUE` etc. I think we should align. I'm not against using the full string, but think constants should be uppercase. Also, what is your preference, `MON` etc or `MONDAY`?

Reply:
I prefer full name, and also I agree constants should be caps but that usually refers the variable name, but fine to change these to caps too. Does what's in the proposal affect this card? If so, should that be completed first and referenced here?

Reply:
The proposal isn't for placeholder but we should be consistent, we don't want one place of runner referencing week days as `MON`, and the other as `monday`. I am happy with the full name, and if you are happy with caps, we should change these to `MONDAY` etc, and I can update the proposal to match.